openmetadata-ui/src/main/resources/ui/src/utils/EntitySummaryPanelUtils.tsx

```diff
@@ -34,109 +37,168 @@ export interface EntityNameProps {
-              <Text className="entity-title text-link-color font-medium m-r-xss">
-                {getTitleName(chart)}
-              </Text>
-              <IconExternalLink width={12} />
-            </div>
-          </Link>
-        ),
-        type: chart.chartType,
-        tags: chart.tags,
-        description: chart.description,
-      }));
-    }
-    case SummaryEntityType.TASK: {
-      return (entityInfo as Task[]).map((task) => ({
-        name: task.name,
-        title: (
-          <Link target="_blank" to={{ pathname: task.sourceUrl }}>
-            <div className="d-flex">
-              <Text
-                className="entity-title text-link-color font-medium m-r-xss"
-                ellipsis={{ tooltip: true }}>
-                {getTitleName(task)}
-              </Text>
-              <IconExternalLink width={12} />
-            </div>
-          </Link>
-        ),
-        type: task.taskType,
-        tags: task.tags,
-        description: task.description,
-      }));
-    }
-    case SummaryEntityType.MLFEATURE: {
-      return (entityInfo as MlFeature[]).map((feature) => ({
-        algorithm: feature.featureAlgorithm,
-        name: feature.name || '--',
-        title: <Text className="entity-title">{getTitleName(feature)}</Text>,
-        type: feature.dataType,
-        tags: feature.tags,
-        description: feature.description,
-      }));
-    }
-    case SummaryEntityType.SCHEMAFIELD: {
-      return (entityInfo as Field[]).map((field) => ({
-        name: field.name,
-        title: <Text className="entity-title"> {getTitleName(field)}</Text>,
-        type: field.dataType,
-        description: field.description,
-        tags: field.tags,
+
+  // sort and highlights list items based on tags and global search highlights data
+  if (Object.values(SummaryEntityType).includes(entityType)) {
+    const tagHighlights = get(highlights, 'tag.name', [] as string[]);
+    const listHighlights: string[] = [];
+    const listHighlightsMap: { [key: string]: number } = {};
+    const SummaryListData = {
+      listItemWithSortOption: [] as BasicEntityInfo[],
+      listItemWithoutSortOption: [] as BasicEntityInfo[],
+    };
+
+    SummaryListHighlightKeys.forEach((highlightKey) => {
+      listHighlights.push(...get(highlights, highlightKey, []));
+    });
+
+    listHighlights?.reduce((acc, colHighlight, index) => {
+      acc[colHighlight.replaceAll(/<\/?span(.*?)>/g, '')] = index;
+
+      return acc;
+    }, listHighlightsMap);
+
+    entityInfo?.reduce((acc, listItem) => {
+      const listItemModifiedData = {
+        name: listItem.name ?? '',
+        title: getTitle({
+          content: getTitleName(listItem),
+          sourceUrl: (listItem as Chart | Task).sourceUrl,
+        }),
+        type:
+          (listItem as Column | Field | MlFeature).dataType ??
+          (listItem as Chart).chartType ??
+          (listItem as Task).taskType,
+        tags: listItem.tags,
```

Comment:
Let's use switch case approach here.
You can refer: https://github.com/open-metadata/OpenMetadata/blob/0ecc3fc136db63b40d7df22ff99b7874fb980aea/openmetadata-ui/src/main/resources/ui/src/utils/DataAssetsHeader.utils.tsx
`getDataAssetsHeaderInfo` method

Reply:
But why we need to add the extra calculation of switch case if it can be done in one line using nullish colescing operator??

Reply:
done